include/envoy/config/ads.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include "envoy/common/exception.h"
+#include "envoy/common/pure.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Config {
+
+class AdsCallbacks {
+public:
+  virtual ~AdsCallbacks() {}
+
+  /**
+   * Called when a configuration update is received.
+   * @param resources vector of fetched resources corresponding to the configuration update.
+   * @throw EnvoyException with reason if the configuration is rejected. Otherwise the configuration
+   *        is accepted. Accepted configurations have their version_info reflected in subsequent
+   *        requests.
+   */
+  virtual void onConfigUpdate(const Protobuf::RepeatedPtrField<ProtobufWkt::Any>& resources) PURE;
```

Comment:
Won't the exception cause Envoy to terminate if bad config or some incorrect config arrives? Is that the intent?

Reply:
This will work (in `AdsApiImpl`) similar to https://github.com/lyft/envoy/blob/master/source/common/config/grpc_subscription_impl.h#L95. I.e. we catch the exception and use that to guide the ACK/NAC of the config.